components/org.wso2.carbon.identity.authz.service/src/main/java/org/wso2/carbon/identity/authz/service/handler/AuthorizationHandler.java

```diff
@@ -75,8 +76,16 @@ public AuthorizationResult handleAuthorization(AuthorizationContext authorizatio
             // If the scopes are configured for the API, it gets the first priority
             if (isScopeValidationRequired(authorizationContext, validateScope)) {
                 validateScopes(authorizationContext, authorizationResult, allowedScopes);
-            } else if (StringUtils.isNotBlank(permissionString) || authorizationContext.getRequiredScopes().size() == 0) {
+            }
+            else if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+                if (StringUtils.isNotBlank(permissionString) || authorizationContext.getRequiredScopes().size() == 0) {
+                    validatePermissions(authorizationResult, user, permissionString, tenantUserRealm);
+                }
+            } else {
+                // TODO: Use OAuth2 scope validator, once merged.
                 validatePermissions(authorizationResult, user, permissionString, tenantUserRealm);
+                // authorizationResult.setAuthorizationStatus(AuthorizationStatus.GRANT);
+
```

Comment:
Use AuthzUtils once https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2206 is merged